src/sql/src/plan/notice.rs

```diff
@@ -102,6 +106,9 @@ impl fmt::Display for PlanNotice {
             PlanNotice::ReplicaDiskOptionDeprecated => {
                 write!(f, "the DISK option is deprecated and has no effect")
             }
+            PlanNotice::KafkaSinkNullablePartitionByFunction => {
+                write!(f, "the PARTITION BY function may evaluate to NULL")
```

Comment:
"PARTITION BY function" -- Is it just a function, or can it be a general expression? If the latter, then better to say "the PARTITION BY expression may evaluate to NULL". (and also above)